flink-connectors/flink-connector-dynamodb/src/main/java/org/apache/flink/streaming/connectors/dynamodb/sink/DynamoDbSinkBuilder.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.connectors.dynamodb.sink;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.connector.base.sink.AsyncSinkBaseBuilder;
+import org.apache.flink.streaming.connectors.dynamodb.config.DynamoDbTablesConfig;
+
+import java.util.Optional;
+import java.util.Properties;
+
+/**
+ * Builder to construct {@link DynamoDbSink}.
+ *
+ * <p>The following example shows the minimum setup to create a {@link DynamoDbSink} that writes
+ * records into DynamoDb
+ *
+ * <pre>{@code
+ * private static class DummyElementConverter
+ *         implements ElementConverter<String, DynamoDbWriteRequest> {
+ *
+ *     @Override
+ *     public DynamoDbWriteRequest apply(String element, SinkWriter.Context context) {
+ *         final Map<String, AttributeValue> map = new HashMap<>();
+ *         map.put("your-key", AttributeValue.builder().s(element).build());
+ *         return new DynamoDbWriteRequest(
+ *                 "your-table-name",
+ *                 WriteRequest.builder()
+ *                         .putRequest(PutRequest.builder().item(map).build())
+ *                         .build());
+ *      }
+ * }
+ * DynamoDbSink<String> dynamoDbSink = DynamoDbSink.<String>builder()
+ *                                          .setElementConverter(new DummyElementConverter())
+ *                                       .build();
+ * }</pre>
+ *
+ * <p>If the following parameters are not set in this builder, the following defaults will be used:
+ *
+ * <ul>
+ *   <li>{@code maxBatchSize} will be 25
+ *   <li>{@code maxInFlightRequests} will be 10
```

Comment:
Shall we set the `maxInFlightRequests` to 50? Since that's the default for the AWS Async Client [...](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/asynchronous.html#:~:text=threads.%20By%20default%2C-,50%20Threads,-are%20generated%20for) .

Reply:
fixed